src/ServiceControl/MessageRedirects/Api/MessageRedirectsModule.cs

```diff
@@ -0,0 +1,144 @@
+﻿namespace ServiceControl.MessageRedirects.Api
+{
+    using System;
+    using System.Linq;
+    using Nancy;
+    using NServiceBus;
+    using Raven.Client;
+    using ServiceBus.Management.Infrastructure.Extensions;
+    using ServiceBus.Management.Infrastructure.Nancy.Modules;
+    using ServiceControl.Infrastructure;
+    using ServiceControl.Infrastructure.Extensions;
+    using ServiceControl.MessageRedirects.InternalMessages;
+
+    public class MessageRedirectsModule : BaseModule
+    {
+        public IBus Bus { get; set; }
+
+        public MessageRedirectsModule()
+        {
+
+            Post["/redirects"] = parameters =>
+            {
+                var message = new CreateMessageRedirect
+                {
+                    FromPhysicalAddress = parameters.fromphysicaladdress,
+                    ToPhysicalAddress = parameters.tophysicaladdress
+                };
+
+                if (string.IsNullOrWhiteSpace(message.FromPhysicalAddress) || string.IsNullOrWhiteSpace(message.ToPhysicalAddress))
+                {
+                    return HttpStatusCode.BadRequest;
+                }
+
+                message.MessageRedirectId = DeterministicGuid.MakeId(message.FromPhysicalAddress, message.ToPhysicalAddress);
+
+                using (var session = Store.OpenSession())
+                {
+                    var redirect = session.Load<MessageRedirect>(MessageRedirect.GetDocumentIdFromMessageRedirectId(message.MessageRedirectId));
```

Comment:
Is this check strictly necessary? If you try to create a redirect and one already exists that does exactly the same thing and they have no identity beyond the two endpoints they connect, what does it matter whether we actually created a new one or just return you the existing one?